examples/static_map/count_by_key_example.cu

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuco/static_map.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/logical.h>
+#include <thrust/transform.h>
+
+#include <cub/block/block_reduce.cuh>
+
+#include <cuda/std/atomic>
+
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <iostream>
+#include <limits>
+
+/**
+ * @file count_by_key_example.cu
+ * @brief Demonstrates usage of the device side APIs for individual operations like insert/find in
+ * the context of a count-by-key operation, i.e. for a histogram over keys.
+ *
+ * Individual operations like a single insert or find can be performed in device code via the
+ * static_map "device_view" types.
+ *
+ * @note This example is for demonstration purposes only. It is not intended to show the most
+ * performant way to do the example algorithm.
+ *
+ */
+
+/**
+ * @brief Inserts keys and counts how often they occur in the input sequence.
+ *
+ * @tparam BlockSize CUDA block size
+ * @tparam Map Type of the map returned from static_map::get_device_mutable_view
+ * @tparam KeyIter Input iterator whose value_type convertible to Map::key_type
+ * @tparam UniqueIter Output iterator whose value_type is convertible to uint64_t
+ *
+ * @param[in] map_view View of the map into which inserts will be performed
+ * @param[in] key_begin The beginning of the range of keys to insert
+ * @param[in] num_keys The total number of keys and values
+ * @param[out] num_unique_keys The total number of distinct keys inserted
+ */
+template <int64_t BlockSize, typename Map, typename KeyIter, typename UniqueIter>
+__global__ void count_by_key(Map map_view,
+                             KeyIter keys,
+                             uint64_t num_keys,
+                             UniqueIter num_unique_keys)
+{
+  typedef cub::BlockReduce<uint64_t, BlockSize> BlockReduce;
```

Comment:
```suggestion
  using BlockReduce = cub::BlockReduce<uint64_t, BlockSize> ;
```
nit: prefer `using` over `typedef`

Reply:
Addressed in [0a8f84a](https://github.com/NVIDIA/cuCollections/pull/280/commits/0a8f84a5fd5ac670727f55f269fb3ff95ba3198b)